app/src/main/java/com/quran/labs/androidquran/ui/SheikhAudioManagerActivity.kt

```diff
@@ -243,8 +244,18 @@ class SheikhAudioManagerActivity : QuranActionBarActivity(), SimpleDownloadListe
             val surah = position + 1
             val downloaded = info.downloadedSuras[surah]
             if (downloaded) {
-              // TODO: show a confirmation dialog before deleting
-              deleteSelection(ArrayList(listOf(surah)))
+              var surahName = quranDisplayData.getSuraName(this, surah, true)
+              val msg = String.format(getString(R.string.audio_manager_remove_audio_msg), surahName)
+              val builder = AlertDialog.Builder(this)
+              builder.setTitle(R.string.audio_manager_remove_audio_title)
+                  .setMessage(msg)
+                  .setPositiveButton(R.string.remove_button
+                  ) { _, _ ->
+                    deleteSelection(ArrayList(listOf(surah)))
+                  }
+                  .setNegativeButton(R.string.cancel
+                  ) { dialog, _ -> dialog.dismiss() }
+              builder.show()
```

Comment:
jazakumAllah khairan - one minor comment - save a reference to the dialog (nullable) and dismiss it `onDestroy` if it's not null to prevent leaks